File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/adapter/RssItemWebViewHolder.java
package de.luhmer.owncloudnewsreader.adapter;

import android.content.SharedPreferences;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.CallSuper;
import androidx.annotation.NonNull;
import androidx.viewbinding.ViewBinding;

import de.luhmer.owncloudnewsreader.async_tasks.RssItemToHtmlTask;
import de.luhmer.owncloudnewsreader.database.model.RssItem;
import de.luhmer.owncloudnewsreader.databinding.SubscriptionDetailListItemWebLayoutBinding;

public class RssItemWebViewHolder extends RssItemViewHolder<SubscriptionDetailListItemWebLayoutBinding> {


    public RssItemWebViewHolder(@NonNull ViewBinding binding, SharedPreferences sharedPreferences) {
        super(binding, sharedPreferences);
    }

    @Override
    protected ImageView getImageViewFavIcon() {
        return binding.layoutThumbnail.imgViewFavIcon;
    }

    @Override
    protected ImageView getStar() {
        return binding.layoutThumbnail.starImageview;
    }

    @Override
    protected ImageView getPlayPausePodcastButton() {
        return binding.layoutThumbnail.podcastWrapper.btnPlayPausePodcast;
    }

    @Override
    protected ImageView getColorFeed() {
        return null;
    }

    @Override
    protected TextView getTextViewTitle() {
        return binding.layoutThumbnail.tvSubscription;
    }

    @Override
    protected TextView getTextViewSummary() {
        return binding.layoutThumbnail.summary;
    }

    @Override
    protected TextView getTextViewBody() {
        return binding.layoutThumbnail.body;
    }

    @Override
    protected TextView getTextViewItemDate() {
        return binding.layoutThumbnail.tvItemDate;
    }

    @Override
    protected FrameLayout getPlayPausePodcastWrapper() {
        return binding.layoutThumbnail.podcastWrapper.flPlayPausePodcastWrapper;
    }

    @Override
    protected ProgressBar getPodcastDownloadProgress() {
        return binding.layoutThumbnail.podcastWrapper.podcastDownloadProgress;
    }

    @CallSuper
    public void bind(@NonNull RssItem rssItem) {
        super.bind(rssItem);

        String htmlPage = RssItemToHtmlTask.getHtmlPage(this.mGlide, rssItem, false, mPrefs, itemView.getContext());
        binding.webViewBody.loadDataWithBaseURL("file:///android_asset/", htmlPage, "text/html", "UTF-8", "");

    }
}

File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/adapter/RssItemWebViewHolder.kt
package de.luhmer.owncloudnewsreader.adapter

import android.content.SharedPreferences
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.annotation.CallSuper
import androidx.viewbinding.ViewBinding
import com.bumptech.glide.RequestManager
import de.luhmer.owncloudnewsreader.async_tasks.RssItemToHtmlTask
import de.luhmer.owncloudnewsreader.database.model.RssItem
import de.luhmer.owncloudnewsreader.databinding.SubscriptionDetailListItemWebLayoutBinding
import de.luhmer.owncloudnewsreader.helper.FavIconHandler

class RssItemWebViewHolder(
    binding: ViewBinding,
    faviconHandler: FavIconHandler,
    glide: RequestManager,
    sharedPreferences: SharedPreferences,
) : RssItemViewHolder<SubscriptionDetailListItemWebLayoutBinding>(
        binding,
        faviconHandler,
        glide,
        sharedPreferences,
    ) {
    override fun getImageViewFavIcon(): ImageView = binding.layoutThumbnail.imgViewFavIcon

    override fun getStar(): ImageView = binding.layoutThumbnail.starImageview

    override fun getPlayPausePodcastButton(): ImageView = binding.layoutThumbnail.podcastWrapper.btnPlayPausePodcast

    override fun getColorFeed(): ImageView? = null

    override fun getTextViewTitle(): TextView = binding.layoutThumbnail.tvSubscription

    override fun getTextViewSummary(): TextView = binding.layoutThumbnail.summary

    override fun getTextViewBody(): TextView = binding.layoutThumbnail.body

    override fun getTextViewItemDate(): TextView = binding.layoutThumbnail.tvItemDate

    override fun getPlayPausePodcastWrapper(): FrameLayout =
        binding.layoutThumbnail.podcastWrapper.flPlayPausePodcastWrapper

    override fun getPodcastDownloadProgress(): ProgressBar =
        binding.layoutThumbnail.podcastWrapper.podcastDownloadProgress

    @CallSuper
    override fun bind(rssItem: RssItem) {
        super.bind(rssItem)
        val htmlPage: String =
            RssItemToHtmlTask.getHtmlPage(
                mGlide,
                rssItem,
                false,
                mPrefs,
                itemView.context,
            )
        binding.webViewBody.loadDataWithBaseURL(
            "file:///android_asset/",
            htmlPage,
            "text/html",
            "UTF-8",
            "",
        )
    }
}
